compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -1858,84 +1864,70 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
           // Do not emit trait impl forwarders with @JavaDefaultMethod
           None
         } else {
-          withScopedVars(
-              mutableLocalVars := mutable.Set.empty,
-              mutatedLocalVars := mutable.Set.empty
-          ) {
-            def isTraitImplForwarder = dd.rhs match {
-              case app: Apply => isImplClass(app.symbol.owner)
-              case _          => false
-            }
-
-            val shouldMarkInline = {
-              sym.hasAnnotation(InlineAnnotationClass) ||
-              sym.name.startsWith(nme.ANON_FUN_NAME) ||
-              adHocInlineMethods.contains(sym.fullName)
-            }
-
-            val shouldMarkNoinline = {
-              sym.hasAnnotation(NoinlineAnnotationClass) &&
-              !isTraitImplForwarder &&
-              !ignoreNoinlineAnnotation(sym)
-            }
+          def isTraitImplForwarder = dd.rhs match {
+            case app: Apply => isImplClass(app.symbol.owner)
+            case _          => false
+          }
 
-            val optimizerHints =
-              OptimizerHints.empty.
-                withInline(shouldMarkInline).
-                withNoinline(shouldMarkNoinline)
+          val shouldMarkInline = {
+            sym.hasAnnotation(InlineAnnotationClass) ||
+            sym.name.startsWith(nme.ANON_FUN_NAME) ||
+            adHocInlineMethods.contains(sym.fullName)
+          }
 
-            val methodDef = {
-              if (sym.isClassConstructor) {
-                val body0 = genStat(rhs)
-                val body1 = {
-                  val needsMove =
-                    isNonNativeJSClass(currentClassSym) && sym.isPrimaryConstructor
+          val shouldMarkNoinline = {
+            sym.hasAnnotation(NoinlineAnnotationClass) &&
+            !isTraitImplForwarder &&
+            !ignoreNoinlineAnnotation(sym)
+          }
 
-                  if (needsMove) moveAllStatementsAfterSuperConstructorCall(body0)
-                  else body0
```

Comment:
Ah, it seems that this change was accidentally included in the "Factor per method body scopedVars into withPerMethodBodyState" commit, whereas it should be in the big commit.

(which, ironically, shows the value of having this separate commit, since I had previously not even noticed that changed ^^)

Reply:
Yes... That's embarrassing :-/